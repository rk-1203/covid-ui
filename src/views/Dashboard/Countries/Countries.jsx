import { useState } from 'react';
import TextField from '@mui/material/TextField';
import MenuItem from '@mui/material/MenuItem';
import Pagination from '@mui/material/Pagination';

import Country from "./Country/Country";
import { COUNTRY_SUMMARY_TABLE_COLUMNS, PAGE_SIZE, SORT_ORDER } from '../../../constants/constant';
import "./Countries.scss";

function Countries({ countries }) {

    const [sortBy, setSortBy] = useState(COUNTRY_SUMMARY_TABLE_COLUMNS[0].id);
    const [sortOrder, setSortOrder] = useState(SORT_ORDER.ASC);
    const [filterName, setFilterName] = useState(COUNTRY_SUMMARY_TABLE_COLUMNS[0].id);
    const [filterValue, setFilterValue] = useState("");
    const [pageNo, setPageNo] = useState(1);

    const handleFilternameChange = (e) => {
        setFilterName(e.target.value);
        setFilterValue("");
        setPageNo(1);
    }

    const handleFilterValueChange = (e) => {
        setFilterValue(e.target.value);
        setPageNo(1);
    }
    
    const handleSortChange = (id) => {
        if(sortBy === id) {
            setSortOrder(order => order === SORT_ORDER.ASC ? SORT_ORDER.DSC : SORT_ORDER.ASC)
        } else {
            setSortBy(id);
            setSortOrder(SORT_ORDER.ASC);
        }
    }

    const filteredCountries = countries
        .filter(country => country[filterName].toString().toLowerCase().includes(filterValue.toString().toLowerCase()))
        .sort((a,b) => {
            if(a[sortBy] < b[sortBy]) { return sortOrder === SORT_ORDER.ASC ? -1 : 1; }
            if(a[sortBy] > b[sortBy]) { return sortOrder === SORT_ORDER.ASC ? 1 : -1; }
            return 0;
        });

    const pageCount = Math.ceil(filteredCountries.length/PAGE_SIZE);

    return (
        <div className="countries-container">
            <div className="header">
                <div className="count">
                    Total Countries : {countries.length}
                </div>
                <div className="filter">
                    <TextField
                        id="outlined-select-currency"
                        select
                        placeholder="Search By" 
                        value={filterName}
                        onChange={handleFilternameChange}
                    >
                        { COUNTRY_SUMMARY_TABLE_COLUMNS.map(({id, displayName}) => (
                            <MenuItem key={id} value={id}>{displayName}</MenuItem>
                        )) }
                    </TextField>
                    <TextField 
                        id="standard-search" 
                        placeholder="Search" 
                        type="search" 
                        variant="filled"
                        value={filterValue}
                        onChange={handleFilterValueChange}
                    />
                </div>
            </div>
            {
                filteredCountries.length > 0 ?
                    <table className="table">
                        <thead>
                            <tr>
                                { 
                                    COUNTRY_SUMMARY_TABLE_COLUMNS.map(({id, displayName}) => (
                                        <th key={id}>
                                            {displayName}
                                            <span className={`material-icons-outlined ${sortBy !== id ? "unselected" : ""}`} onClick={() => handleSortChange(id)}>
                                                {sortOrder === SORT_ORDER.ASC ? "expand_less":"expand_more"}
                                            </span>
                                        </th>
                                    )) 
                                }
                            </tr>
                        </thead>
                        <tbody>
                            { 
                                filteredCountries
                                    .slice((pageNo-1)*PAGE_SIZE, pageNo*PAGE_SIZE)
                                    .map(country => <Country key={country.ID} {...country}/>) 
                            }
                        </tbody>
                    </table> :
                    <div className="no-results">No Result Found</div>
            }
            {
                pageCount > 1 &&
                <Pagination 
                    count={pageCount} 
                    color="primary" 
                    className="pagination" 
                    page={pageNo} 
                    onChange={(_, pageNo) => setPageNo(pageNo)}
                />
            }
        </div>
    );
}

export default Countries;
