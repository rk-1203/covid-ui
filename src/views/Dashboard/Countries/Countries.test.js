import React from 'react';
import { act, getByRole, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Countries from "./Countries";

const props = {
    countries: [
        {
            ID: "e0296b5d-0c21-484d-be0f-7ee48a5f9a7b",
            Country: "Afghanistan",
            CountryCode: "AF",
            Slug: "afghanistan",
            NewConfirmed: 0,
            TotalConfirmed: 167739,
            NewDeaths: 0,
            TotalDeaths: 7454,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "a51224c1-1c0c-4de0-997e-5154d146a60a",
            Country: "Albania",
            CountryCode: "AL",
            Slug: "albania",
            NewConfirmed: 0,
            TotalConfirmed: 265716,
            NewDeaths: 0,
            TotalDeaths: 3393,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "2a89dd4b-6f9c-4aa7-809e-274d82887084",
            Country: "Algeria",
            CountryCode: "DZ",
            Slug: "algeria",
            NewConfirmed: 0,
            TotalConfirmed: 259088,
            NewDeaths: 0,
            TotalDeaths: 6667,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "3416b50c-079e-48a4-a6fb-b56da7ac3cbd",
            Country: "Andorra",
            CountryCode: "AD",
            Slug: "andorra",
            NewConfirmed: 0,
            TotalConfirmed: 36808,
            NewDeaths: 0,
            TotalDeaths: 147,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "51449e6d-34a7-4fb1-bc05-15b41af6d823",
            Country: "Angola",
            CountryCode: "AO",
            Slug: "angola",
            NewConfirmed: 0,
            TotalConfirmed: 98424,
            NewDeaths: 0,
            TotalDeaths: 1896,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "72468853-06cc-4ad3-8aee-316d1092f99d",
            Country: "Antigua and Barbuda",
            CountryCode: "AG",
            Slug: "antigua-and-barbuda",
            NewConfirmed: 0,
            TotalConfirmed: 6853,
            NewDeaths: 0,
            TotalDeaths: 131,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "b5f6b7ab-5818-4e54-883b-5f096c7a3ade",
            Country: "Argentina",
            CountryCode: "AR",
            Slug: "argentina",
            NewConfirmed: 0,
            TotalConfirmed: 8648075,
            NewDeaths: 0,
            TotalDeaths: 123227,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "dad9b03d-acdc-41c5-86a2-164db926c6b8",
            Country: "Armenia",
            CountryCode: "AM",
            Slug: "armenia",
            NewConfirmed: 0,
            TotalConfirmed: 394074,
            NewDeaths: 0,
            TotalDeaths: 8102,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "a1ac89f7-1891-4e45-af78-dacdd4d01ff2",
            Country: "Australia",
            CountryCode: "AU",
            Slug: "australia",
            NewConfirmed: 31510,
            TotalConfirmed: 2811390,
            NewDeaths: 71,
            TotalDeaths: 4373,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "d8317184-6f56-4fef-b391-c8de386486f5",
            Country: "Austria",
            CountryCode: "AT",
            Slug: "austria",
            NewConfirmed: 0,
            TotalConfirmed: 2111314,
            NewDeaths: 0,
            TotalDeaths: 14271,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        },
        {
            ID: "796698c5-d73d-41c3-8722-d0f2b86e98ec",
            Country: "Azerbaijan",
            CountryCode: "AZ",
            Slug: "azerbaijan",
            NewConfirmed: 0,
            TotalConfirmed: 710366,
            NewDeaths: 0,
            TotalDeaths: 8920,
            NewRecovered: 0,
            TotalRecovered: 0,
            Date: "2022-02-09T16:49:33.508Z",
            Premium: {}
        }
    ],
}

describe("Countries", () => {

    it("renders properly", () => {
        const { container } = render(<Countries {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it("handles filters and pagination properly", async () => {
        const { container,  getByPlaceholderText, getByText, getByRole } = render(<Countries {...props} />);
        const searchElement = getByPlaceholderText("Search");
        const filterByElement = getByPlaceholderText("Search By").previousSibling;
        const page1ButtonElement = getByRole('button', {name: /1/});
        expect(page1ButtonElement).toBeInTheDocument();
        act(() => userEvent.type(searchElement, '-1'));
        await waitFor(() => expect(page1ButtonElement).not.toBeInTheDocument());
        userEvent.click(filterByElement);
        const optionElement = getByText("Total Deaths");
        act(() => userEvent.click(optionElement));
        expect(searchElement.value).toBe("");
        expect(container.firstChild).toMatchSnapshot();
    });

})