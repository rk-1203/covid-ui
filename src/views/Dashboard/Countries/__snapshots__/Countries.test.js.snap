// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries handles filters and pagination properly 1`] = `
<div
  class="countries-container"
>
  <div
    class="header"
  >
    <div
      class="count"
    >
      Total Countries : 
      11
    </div>
    <div
      class="filter"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="outlined-select-currency"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-select-currency"
            role="button"
            tabindex="0"
          >
            Total Deaths
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            placeholder="Search By"
            tabindex="-1"
            value="TotalDeaths"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1lfthva-MuiInputBase-input-MuiFilledInput-input"
            id="standard-search"
            placeholder="Search"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Country
          <span
            class="material-icons-outlined "
          >
            expand_less
          </span>
        </th>
        <th>
          New Confirmed
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Confirmed
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          New Deaths
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Deaths
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          New Recovered
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Recovered
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Afghanistan
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          1,67,739
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          7,454
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Albania
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          2,65,716
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          3,393
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Algeria
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          2,59,088
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          6,667
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Andorra
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          36,808
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          147
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Angola
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          98,424
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          1,896
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Antigua and Barbuda
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          6,853
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          131
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Argentina
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          86,48,075
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          1,23,227
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Armenia
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          3,94,074
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          8,102
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Australia
        </td>
        <td
          data-th="New Confirmed"
        >
          31,510
        </td>
        <td
          data-th="Total Confirmed"
        >
          28,11,390
        </td>
        <td
          data-th="New Deaths"
        >
          71
        </td>
        <td
          data-th="Total Deaths"
        >
          4,373
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Austria
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          21,11,314
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          14,271
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-text pagination css-1oj2twp-MuiPagination-root"
  >
    <ul
      class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Countries renders properly 1`] = `
<div
  class="countries-container"
>
  <div
    class="header"
  >
    <div
      class="count"
    >
      Total Countries : 
      11
    </div>
    <div
      class="filter"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="outlined-select-currency"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="outlined-select-currency"
            role="button"
            tabindex="0"
          >
            Country
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            placeholder="Search By"
            tabindex="-1"
            value="Country"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputTypeSearch css-1lfthva-MuiInputBase-input-MuiFilledInput-input"
            id="standard-search"
            placeholder="Search"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          Country
          <span
            class="material-icons-outlined "
          >
            expand_less
          </span>
        </th>
        <th>
          New Confirmed
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Confirmed
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          New Deaths
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Deaths
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          New Recovered
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
        <th>
          Total Recovered
          <span
            class="material-icons-outlined unselected"
          >
            expand_less
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Afghanistan
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          1,67,739
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          7,454
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Albania
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          2,65,716
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          3,393
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Algeria
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          2,59,088
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          6,667
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Andorra
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          36,808
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          147
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Angola
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          98,424
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          1,896
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Antigua and Barbuda
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          6,853
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          131
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Argentina
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          86,48,075
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          1,23,227
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Armenia
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          3,94,074
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          8,102
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Australia
        </td>
        <td
          data-th="New Confirmed"
        >
          31,510
        </td>
        <td
          data-th="Total Confirmed"
        >
          28,11,390
        </td>
        <td
          data-th="New Deaths"
        >
          71
        </td>
        <td
          data-th="Total Deaths"
        >
          4,373
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
      <tr
        class="country-row"
      >
        <td
          data-th="Country"
        >
          Austria
        </td>
        <td
          data-th="New Confirmed"
        >
          0
        </td>
        <td
          data-th="Total Confirmed"
        >
          21,11,314
        </td>
        <td
          data-th="New Deaths"
        >
          0
        </td>
        <td
          data-th="Total Deaths"
        >
          14,271
        </td>
        <td
          data-th="New Recovered"
        >
          0
        </td>
        <td
          data-th="Total Recovered"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-text pagination css-1oj2twp-MuiPagination-root"
  >
    <ul
      class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
